packages/blitz-rpc/src/data-client/rpc.ts

```diff
@@ -166,18 +152,12 @@ export function __internal_buildRpcClient({
             err.stack = null!
             throw err
           }
-
           if (payload.error) {
             let error = deserialize({
               json: payload.error,
               meta: payload.meta?.error,
             }) as any
-            // We don't clear the publicDataStore for anonymous users,
-            // because there is not sensitive data
-            if (error.name === "AuthenticationError" && getPublicDataStore().getData().userId) {
-              getPublicDataStore().clear()
-            }
-
+            await Promise.all(__BLITZ_onRpcError(error))
```

Comment:
we have mixture here. onSessionCreated is using `document.dispatchEvent` but `onRpcError` does not use document event system and instead just calls the global handler.

We should make this consistent. I'm not sure which is the best way, but I'm leaning towards the document.dispatchEvent way, because it seems more decoupled.

Reply:
this function is not ran on each render. It's only called when react-query triggers a fetch.

also since this rpcError is an event, we don't have to wait for it to be processed. Which is why dispatchEvent makes sense to me.

Reply:
Ah okay, I have updated the implementation using `CustomEvent`